srdt_analysis/chunker.py

```diff
@@ -31,11 +42,26 @@ def split_markdown(self, markdown: str) -> List[SplitDocument]:
         documents = self._character_recursive_splitter.split_documents(md_header_splits)
         return [SplitDocument(doc.page_content, doc.metadata) for doc in documents]
 
+    def split_html(self, html: str) -> List[SplitDocument]:
+        html_header_splits = self._html_splitter.split_text(html)
+        documents = self._character_recursive_splitter.split_documents(
+            html_header_splits
+        )
+        return [SplitDocument(doc.page_content, doc.metadata) for doc in documents]
+
     def split_character_recursive(self, content: str) -> List[SplitDocument]:
         text_splits = self._character_recursive_splitter.split_text(content)
         return [SplitDocument(text, {}) for text in text_splits]
 
-    def split(self, content: str, content_type: str = "markdown"):
+    def split(
+        self,
+        content: str,
+        content_type: ChunkerContentType,
+    ) -> List[SplitDocument]:
         if content_type.lower() == "markdown":
             return self.split_markdown(content)
+        if content_type.lower() == "html":
+            return self.split_html(content)
+        if content_type.lower() == "character_recursive":
+            return self.split_character_recursive(content)
         raise ValueError(f"Unsupported content type: {content_type}")
```

Comment:
Potentiellement tu peux faire comme ça aussi. Perso je trouve que c'est un peu plus clean parce que ça évite de répéter des conditions et on appelle `content_type.lower()` qu'une seule fois :
```suggestion
        content_type_to_splitters = {
            "markdown": self.split_markdown,
            "html": self.split_html,
            "character_recursive": self.split_character_recursive,
        }
        splitter_func = content_type_to_splitters.get(content_type.lower())
        if splitter_func is None:
            raise ValueError(f"Unsupported content type: {content_type}")
        return splitter_func(content)
```

Que tu prennes la suggestion ou pas, tu peux mettre `content_type.lower()` dans une variable pour éviter de l'appeller plusieurs fois :

```suggestion
        content_type_lower = content_type.lower()
        if content_type_lower == "markdown":
            return self.split_markdown(content)
        if content_type_lower == "html":
            return self.split_html(content)
        if content_type_lower == "character_recursive":
            return self.split_character_recursive(content)
```

Reply:
Yes, bonne sugestion, si je fais ça, il faut que je type chaque type de contenu afin d'éviter d'avoir des erreurs les keys. Je vois comment faire ça pour avoir les types au passage